From 0971f3570c35855f0b1f78bd88489e5dbae4d3df Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 15 Nov 2018 18:14:44 +0200
Subject: [PATCH 0793/1051] Revert "net: mvpp2: flow control tx stop queue only
 on matched cpu"

This reverts commit d494984fecce8c2ffc987dfb1304666a8f42bc7b.
"net: mvpp2: flow control tx stop queue only on matched cpu"

The reverted commit is replaced by patch
"net: mvpp2: fix tx stop-wake queue for combined cpu-txq map"

Change-Id: Id6b20b91f209907ee2cee49030e0d47fdd8b48cb
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60973
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index b68edeb8fc06..e47619a22f12 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -3821,9 +3821,7 @@ static netdev_tx_t mvpp2_tx(struct sk_buff *skb, struct net_device *dev)
 			mvpp2_aggr_txq_pend_desc_add(port, frags);
 		}
 
-		if (unlikely(txq_pcpu->count >= txq_pcpu->stop_threshold) &&
-		    thread == txq_id) {
-			/* Don't stop TXQ not related to this cpu/thread */
+		if (unlikely(txq_pcpu->count >= txq_pcpu->stop_threshold)) {
 			nq = netdev_get_tx_queue(dev, txq_id);
 			netif_tx_stop_queue(nq);
 		}
-- 
2.17.1

